src/mito2/src/worker/handle_open.rs

```diff
@@ -59,10 +59,34 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             return RegionNotFoundSnafu { region_id }.fail();
         }
 
+        Ok(())
+    }
+
+    pub(crate) async fn handle_open_request(
+        &mut self,
+        region_id: RegionId,
+        request: RegionOpenRequest,
+        sender: OptionOutputTx,
+    ) {
+        let Some(sender) = self
+            .opening_regions
+            .wait_for_opening_region(region_id, sender)
+        else {
+            return;
+        };
+
+        if self.regions.is_region_exists(region_id) {
+            sender.send(Ok(0));
+            return;
+        }
```

Comment:
prefer to check existence first.
```suggestion
        if self.regions.is_region_exists(region_id) {
            sender.send(Ok(0));
            return;
        }

        let Some(sender) = self
            .opening_regions
            .wait_for_opening_region(region_id, sender)
        else {
            return;
        };
```